src/main/java/com/joelj/jenkins/eztemplates/jobtypes/JobsFacade.java

```diff
@@ -0,0 +1,113 @@
+package com.joelj.jenkins.eztemplates.jobtypes;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.kohsuke.stapler.Ancestor;
+
+import com.google.common.base.Throwables;
+import com.joelj.jenkins.eztemplates.exclusion.Exclusions;
+
+import hudson.model.AbstractProject;
+import hudson.model.Job;
+import hudson.model.Label;
+import hudson.scm.SCM;
+import hudson.triggers.Trigger;
+import hudson.triggers.TriggerDescriptor;
+
+
+/**
+ * Facade to encapsulate the access to different types of Jenkins Jobs. Should be used when direct use of Job isn't possible and you have
+ * to deal with concrete implementations of Jobs.
+ *
+ * @see <a href="https://github.com/jenkinsci/pipeline-plugin/blob/master/DEVGUIDE.md#historical-background">pipeline's DEVGUIDE.md#historical-background</a>
+ * @author Juan Pablo Santos Rodríguez
+ */
+public class JobsFacade {
+
+    private static boolean pipelinesPluginEnabled = Exclusions.checkPlugin("workflow-job") == null;
+
+    /**
+     * Verifies if the the plugin applies to the Jenkins job type.
+     *
+     * @param jobType Jenkins job type.
+     * @return {@code true} if it is either an {@link AbstractProject} or a {@link WorkflowJob}.
+     */
+    public static boolean isPluginApplicableTo( Class< ? extends Job > jobType ) {
+        if( pipelinesPluginEnabled ) {
+            return AbstractProject.class.isAssignableFrom( jobType )
+                || WorkflowJob.class.isAssignableFrom( jobType );

```

Comment:
I mean, if we find a WorkflowJob - shouldn't we attempt to proceed anyway? The fact we have an instance of it means the code is still resident.